src/firmware/app/control/trajectory_planner_test.cpp

```diff
@@ -0,0 +1,529 @@
+        velocity.push_back(inital_speed);
+
+        for (unsigned i = 1; i < trajectory->num_elements - 1; i++)
+        {
+            const float dx = trajectory->trajectory_elements[i + 1].position.x -
+                             trajectory->trajectory_elements[i].position.x;
+            const float dy = trajectory->trajectory_elements[i + 1].position.y -
+                             trajectory->trajectory_elements[i].position.y;
+
+            const float dt = trajectory->trajectory_elements[i + 1].time -
+                             trajectory->trajectory_elements[i].time;
+
+            velocity.push_back(sqrt((pow(dx, 2) + pow(dy, 2))) / dt);
+        }
+        return velocity;
+    }
+
+    static std::vector<float> getAccelerationsFromSpeed(std::vector<float> speeds,
+                                                        Trajectory_t* trajectory)
+    {
+        std::vector<float> acceleration;
+
+        // Calculate the delta-position and the delta-time
+        acceleration.push_back(0.0);
+
+        for (unsigned i = 1; i < trajectory->num_elements - 2; i++)
+        {
+            const float dv = speeds[i + 1] - speeds[i];
+            const float dt = trajectory->trajectory_elements[i + 1].time -
+                             trajectory->trajectory_elements[i].time;
+
+            acceleration.push_back(dv / dt);
+        }
+        return acceleration;
+    }
+};
+
+TEST_F(TrajectoryPlannerTest, check_trajectory_length_and_end_points_match_path)
+{
+    Polynomial2dOrder3_t path = {
+        .x = {.coefficients = {2, 0, 1, 0}},
+        .y = {.coefficients = {1, 0, 1, 0}},
+
+    };
+
+    FirmwareRobotPathParameters_t path_parameters;
+    path_parameters.path                       = path;
+    path_parameters.t_start                    = 0;
+    path_parameters.t_end                      = 1;
+    path_parameters.num_segments               = 5999;
+    path_parameters.max_allowable_acceleration = 3;
+    path_parameters.max_allowable_speed        = 3;
+    path_parameters.initial_speed              = 0;
+    path_parameters.final_speed                = 0;
+
+    TrajectoryElement_t const_arc_elements[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    Trajectory_t trajectory = {.trajectory_elements = const_arc_elements};
+
+    app_trajectory_planner_generateConstantArcLengthSegmentation(path_parameters,
+                                                                 &trajectory);
+
+    // Create the parmeterization to contain the desired number of segments
+    CREATE_STATIC_ARC_LENGTH_PARAMETRIZATION(arc_length_param,
+                                             TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS);
+    // Get all of the points for the arc length parameterization (Not constant arc length
+    // segments)
+    shared_polynomial_getArcLengthParametrizationOrder3(
+        path_parameters.path, path_parameters.t_start, path_parameters.t_end,
+        arc_length_param);
+
+    const float arc_segment_length =
```

Comment:
Why not use the `shared_polynomial2d_getTotalArcLength` function here? Is that not the same?

Reply:
Changed to the polynimial2d function